cmd/registry/conformance/conformance-task.go

```diff
@@ -107,40 +110,77 @@ func (task *ComputeConformanceTask) Run(ctx context.Context) error {
 		return err
 	}
 
-	// Generate conformance report
+	// Run the linters and compute conformance report
 	conformanceReport := initializeConformanceReport(task.Spec.GetName(), task.StyleguideId)
-	err = task.computeConformanceReport(ctx, root, conformanceReport)
-	if err != nil {
-		log.Errorf(ctx,
-			"Failed to compute the conformance for spec %s: %s",
-			task.Spec.GetName(),
-			err,
-		)
-		return err
+	guidelineReportsMap := make(map[string]int)
+	for _, metadata := range task.LintersMetadata {
+
+		linterResponse, err := task.invokeLinter(ctx, root, metadata)
+		// If a linter returned an error, we shouldn't stop linting completely across all linters and
+		// discard the conformance report for this spec. We should log but still continue, because there
+		// may still be useful information from other linters that we may be discarding.
+		if err != nil {
+			log.Errorf(ctx, "Linter error: %s", err)
+			continue
+		}
+
+		task.computeConformanceReport(ctx, conformanceReport, guidelineReportsMap, linterResponse, metadata)
 	}
 
 	return task.storeConformanceReport(ctx, conformanceReport)
 }
 
-func (task *ComputeConformanceTask) computeConformanceReport(
+func (task *ComputeConformanceTask) invokeLinter(
```

Comment:
Moved `invokeLinter` from `linter.go` to here. Making invokeLinter a function of `ComputeConformanceTask` will open the possibility to mock this function while testing the `Run()` function of this task.

Reply:
I might be mistaken about my understanding of mocking in golang,  it works differently than my understanding from python background.

Reply:
[This](https://github.com/apigee/registry/pull/454) is my plan for testing `exec.Command` usage in. our codebase.  
The PR. is still WIP, it currently contains tests for package `controller`, I am planning to add tests for `invokeLinter` in the similar way.

We can defer this discussion of testing `invokeLinter()` to that PR.